app/scripts/models/reliers/oauth.js

```diff
@@ -140,10 +146,10 @@ define(function (require, exports, module) {
       self._importRequiredSearchParam('client_id', 'clientId');
       self._importRequiredSearchParam('scope', 'scope');
       self.importSearchParam('state');
-      self.importSearchParam('redirect_uri', 'redirectUri');
+      self._importValidateOptionalSearchParam('redirect_uri', 'redirectUri', Validate.isUriValid, OAuthErrors);
```

Comment:
Agreed, I will add validation for line 174 for now. Apparently Joi for browser could be between [32KB - 128KB](https://github.com/hapijs/joi/issues/528) gzipped. Not sure if that is a deal breaker.